fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -498,11 +507,10 @@ public Response createOrReplaceObjectRdf(
             }
 
             final var providedContentType = getSimpleContentType(requestContentType);
-
             final var created = new AtomicBoolean(false);
 
-                if ((resourceExists && resource() instanceof Binary) ||
-                        (!resourceExists && isBinary(interactionModel,
+            if ((resourceExists && (isBinaryTombstone(overwriteTombstone) || resource() instanceof Binary)) ||
```

Comment:
I think the use of `resource()` everywhere made me want to stick with that, but overall I see what you're saying. I'll see about refactoring a bit so that we can use `resource` so that we don't need to re-check for conditions we've already met.